modules/xfem/src/userobjects/GeometricCutUserObject.C

```diff
@@ -17,13 +17,16 @@
 #include "XFEMAppTypes.h"
 #include "EFAElement2D.h"
 #include "EFAElement3D.h"
+#include "XFEMElementPairLocator.h"
+#include "DisplacedProblem.h"
 
 template <>
 InputParameters
 validParams<GeometricCutUserObject>()
 {
   InputParameters params = validParams<CrackFrontPointsProvider>();
   params.addClassDescription("Base UserObject class for XFEM Geometric Cuts");
+  params.addParam<bool>("heal_mesh", false, "Heal previous cuts at every time step");
```

Comment:
Maybe rename to `always_heal_mesh`. Rename the member variable as well.  The reason I suggest this is that derived classes may want to have their own more fine-grained control over healing, and the name `heal_mesh` might cause confusion with the options for that, which will be added in the future.

Reply:
`heal_every_time` would work too.

Reply:
`heal_always`